### PR TITLE
Use an incremental group id when creating a group in contracts

### LINKED ISSUE #682: createGroup can be frontrun
A malicious actor can monitor calls of createGroup(groupId, admin) and frontrun it with createGroup(groupId, admin1). This prevents new group's creation. If we adopt an incremental groupId, once groupId increases by one each time createGroup is called, we can prevent this from happening

https://github.com/semaphore-protocol/semaphore/blob/8eb19e83fda62644872b2fcfbd85011d3b2c21e2/packages/contracts/contracts/Semaphore.sol#L27
https://github.com/semaphore-protocol/semaphore/blob/8eb19e83fda62644872b2fcfbd85011d3b2c21e2/packages/contracts/contracts/Semaphore.sol#L34

### PR DESCRIPTION
**Pull request by vplasencia**
## Related Issue(s)

Closes #682 

## Checklist

-   [x] My code follows the style guidelines of this project
-   [x] I have performed a self-review of my code
-   [x] I have commented my code, particularly in hard-to-understand areas
-   [ ] I have made corresponding changes to the documentation
-   [x] My changes generate no new warnings
-   [x] I have run `yarn prettier` and `yarn lint` without getting any errors
-   [x] I have added tests that prove my fix is effective or that my feature works
-   [x] New and existing unit tests pass locally with my changes
-   [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: packages/contracts/contracts/Semaphore.sol [hunk 1]
```diff
@@ -17,21 +17,27 @@ contract Semaphore is ISemaphore, SemaphoreGroups {
...
     mapping(uint256 => Group) public groups;
 
+    /// @dev Counter to assign the group id.
+    uint256 public groupCounter;
+
     /// @dev Initializes the Semaphore verifier used to verify the user's ZK proofs.
     /// @param _verifier: Semaphore verifier addresse.
     constructor(ISemaphoreVerifier _verifier) {
         verifier = _verifier;
+        groupCounter = 0;
```
Comment (cedoor) [inline packages/contracts/contracts/Semaphore.sol]: This shouldn't be necessary. `groupCounter` is 0 by default.

File: packages/contracts/contracts/Semaphore.sol [hunk 2]
```diff
@@ -17,21 +17,27 @@ contract Semaphore is ISemaphore, SemaphoreGroups {
     /// @dev Gets a group id and returns the group parameters.
     mapping(uint256 => Group) public groups;
 
+    /// @dev Counter to assign the group id.
```
Comment (cedoor) [inline packages/contracts/contracts/Semaphore.sol]:
```suggestion
    /// @dev Counter to assign an incremental id to the groups.
```

File: packages/contracts/.prettierignore
Comment (cedoor) [inline packages/contracts/.prettierignore:1]: Why do we need this separate `.prettierignore` file?

**Review by cedoor — CHANGES_REQUESTED** [review]

File: packages/contracts/.prettierignore
Comment (vplasencia) [reply packages/contracts/.prettierignore:1]: Thanks! I updated the prettier config in contracts.

**Review by cedoor — APPROVED** [review]
LGTM great!!

Side note, we could add a new `createGroup` function without any parameter, where the admin is msg.sender.

**Comment by vplasencia** [comment]
Hey @cedoor I think that's a great idea, we could create a new issue. This will make it easier to use the function. There are a lot of cases where the admin is `msg.sender`.

**Review by 0xDatapunk — APPROVED** [review]

**Review by 0xbok — APPROVED** [review]

**Pull request merged**